### ISSUE TITLE
Bug: [no-magic-numbers] false positive, default values for class members are not ignored by ignoreDefaultValues

### ISSUE THREAD

**Issue by Alexnortung** [eslint/eslint#16361]
### Environment

Node version: 16.16.0
npm version: 8.11.0
Local ESLint version: 8.24.0
Operating System: Linux

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "@typescript-eslint/no-magic-numbers": ["error", {
      "ignoreDefaultValues": true
    }]
  }
}
```
</details>

```js
/* eslint no-magic-numbers: ["error", { "ignoreDefaultValues": true }] */
class Anim {
  duration = 200;
}

new Anim();
```

### What did you expect to happen?

I expected that the initializer for the field for the class would not be a magic number.
However this might not be a "default" value, but an initializer, but I would expect it to not report an error anyway.

### What actually happened?

I got an error from eslint.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This issue was first posted on typescript-eslint/typescript-eslint#5695

**Comment by zhatongning** [comment]
`duration = 200` is an **PropertyDefinition** type 。And from [the source code of the migic number rule](https://github.com/eslint/eslint/blob/1cc4b3a8f82a7945dcd8c59550b6a906a0fabbb4/lib/rules/no-magic-numbers.js#L103), the option `ignoreDefaultValues` only works with `AssignmentPattern` which  you could find in Default function parameters ast.
